examples/kalman_filter.ipynb

```diff
@@ -0,0 +1,490 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Kalman Filter"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This example optimizes the parameters of a [Kalman-Filter](https://en.wikipedia.org/wiki/Kalman_filter).\n",
+    "\n",
+    "This example is available as a Jupyter notebook [here](https://github.com/patrick-kidger/diffrax/blob/main/examples/kalman_filter.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import functools as ft\n",
+    "from types import SimpleNamespace\n",
+    "from typing import Optional\n",
+    "\n",
+    "import diffrax as dfx\n",
+    "import equinox as eqx  # https://github.com/patrick-kidger/equinox\n",
+    "import jax.numpy as jnp\n",
+    "import jax.random as jr\n",
+    "import jax.tree_util as jtu\n",
+    "import matplotlib.pyplot as plt\n",
+    "import optax  # https://github.com/deepmind/optax"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We use [Equinox](https://github.com/patrick-kidger/equinox) to build the Kalman-Filter implementation and to represent linear, time-invariant systems (LTI systems). \n",
+    "\n",
+    "We use [Optax](https://github.com/deepmind/optax) for optimisers (Adam etc.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem Formulation\n",
+    "\n",
+    "Assume that there exists some *unknown* dynamical system of the form\n",
+    "\n",
+    "$\\frac{dx}{dt} = f(x(t), u(t), t)$\n",
+    "\n",
+    "$y(t) = g(x(t)) + \\epsilon$ \n",
+    "\n",
+    "where \n",
+    "- $u(t)$ denotes the time-dependent input to the system\n",
+    "- $y(t)$ denotes the time-dependent output / measurement to the system\n",
+    "- $x(t)$ denotes the time-dependent state of the system (which is not directly measureable in general)\n",
+    "- $f,g$ denote the time-dependent dynamics- and measurement-function, respectively\n",
+    "- $\\epsilon$ denotes random measurement uncertainty\n",
+    "\n",
+    "*The goal is to infer $x(t)$ from $y(1:t)$ even though $f,g,\\epsilon$ are unkown.*\n",
```

Comment:
I don't think `y(1:t)` is standard notation. I think it'd be technically correct to write "The goal is to infer $x$ from $y$ even though", or to use $\\{x(t)\\}_t$ and $\\{y(t)\\}_t$ if one wished to emphasise the time-dependence.

Reply:
Hmm, so i just wanted to highlight that we are doing filtering and not smoothing, i.e. not $x(1:T)$ from $y(1:T)$. 

But these are again maybe just details and then often simpler is better. So, sure!